### PR TITLE
adds verifier for identity fields.

### LINKED ISSUE #576: Validation: Ash.Resource.Verifiers.VerifyIdentityFields
**Is your feature request related to a problem? Please describe.**

Given this schema:
```elixir
  attributes do
    uuid_primary_key :id

    attribute :title, :string do
      allow_nil? false
    end

    attribute :slug, :string do
      allow_nil? false
    end

    attribute :content, :string do
      allow_nil? false
    end

    create_timestamp(:inserted_at)
    update_timestamp(:updated_at)
  end

 relationships do
    belongs_to :site, MyApp.Sites.Site
  end

  identities do
    identity :slug_site, [:slug, :site] # <-- Should be :site_id
  end
```

Since `:site` is a relationship, this would silently fail to enforce identity constraints.

**Describe the solution you'd like**

At least the case of relationship identities could be validated at compile time in a new `Ash.Resource.Verifiers.VerifyIdentityFields` validator module and emit a warning (or perhaps error?) at compile time. There may be other scenarios that could be checked, but this stands out an footgun that should be caught automatically.

### PR DESCRIPTION
**Pull request by TwistingTwists**
* adds verifier for identity fields.
* raises error if verifier fails
* tests added

Closes #576.

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [x] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by zachdaniel** [comment]
I think a simpler check here would be a bit better. Specifically “is this field an attribute, calculation, or aggregate”. That would cover the case of using a relationship name in addition to cases like typos or things like that.

File: lib/ash/resource/verifiers/verfiy_identities.ex [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  def verify(dsl) do
+    identities = Ash.Resource.Info.identities(dsl)
+
+    for identity <- identities do
+      for key <- identity.keys do
+        if !Ash.Resource.Info.attribute(dsl, key) do
+          raise Spark.Error.DslError,
+            module: Spark.Dsl.Verifier.get_persisted(dsl, :module),
+            message: "no such attribute #{inspect(key)}"
```
Comment (zachdaniel) [inline lib/ash/resource/verifiers/verfiy_identities.ex:15]: lets add a `path` to the error, `path: [:identities, identity.name]`.

Additionally, let's make the error message a bit clearer: "All identity keys must be attributes. Got: #{inspect(key)}"

**Review by zachdaniel — APPROVED** [review]

File: lib/ash/resource/verifiers/verfiy_identities.ex (hunk 1 above)
Comment (TwistingTwists) [reply lib/ash/resource/verifiers/verfiy_identities.ex]: done.

**Review by zachdaniel — APPROVED** [review]

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**